roles/nginx/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+---
+- name: import global variables
+  include_vars:
+    dir: "{{ inventory_dir }}/group_vars/"
+
+- name: install/upgrade nginx
+  package:
+    name: nginx
+    state: latest
+
+- name: configure nginx
+  template:
+    src: "files/nginx.conf"
+    dest: "/etc/nginx/nginx.conf"
+    serole: object_r
+    setype: httpd_config_t
+    seuser: system_u
+  notify: restart nginx
+
+- name: add firewall rules for http/https
+  firewalld:
+    service: "{{ item }}"
+    immediate: yes
+    permanent: true
+    state: enabled
+  loop:
+    - http
+    - https
+
+- name: create group for web development permissions
+  group: name=www-admin
+
+- name: create system user for web development permissions
+  user:
+    comment: "system user for managing website deployments - do not use"
+    create_home: no
+    group: www-admin
+    home: "/var/www"
+    name: www-admin
+    system: yes
+
+- name: add authorized users to web development group
+  user:
+    append: yes
+    groups: www-admin
+    name: "{{ item }}"
+  loop: "{{ authorized_users }}"
+
+- name: create root web directory (/var/www/)
+  file:
+    state: directory
+    path: "/var/www/"
+    mode: 0755
+    setype: httpd_sys_content_t
+    seuser: system_u
+
+- name: create default server block root directory
+  file:
+    state: directory
+    path: "/var/www/html/"
+    mode: 0755
+    group: www-admin
+    owner: www-admin
+    setype: httpd_sys_content_t
+    seuser: system_u
+
+- name: copy default html pages
+  copy:
+    src: "files/html/{{ item }}"
+    dest: "/var/www/html/{{ item }}"
+    mode: 0644
+    group: www-admin
+    owner: www-admin
+    setype: httpd_sys_content_t
+  loop:
+    - 404.html
+    - 50x.html
+    - index.html
+    - nginx-logo.png
+    - poweredby.png
+
```

Comment:
So we just pull the actual files from the repo into /roles/nginx/files/ and then iterate through them huh?
I get a feeling this will be a drag when we have to add all of the website files we were given.

Reply:
@BonjeBongo These are for the default NGINX pages. Since it was only five, I did it this way to make it easier the individual five files. For a larger website, it makes more sense to package it up as a ZIP, push it somewhere public, and use the `unarchive` module to unpack the ZIP file. Typing out each file name would be _very_ tedious. :sweat_smile: